src/mewbot/api/registry.py

```diff
@@ -44,6 +44,7 @@ def __call__(  # type: ignore
             raise TypeError("Attempting to create a non registrable class")
 
         obj: Any = cls.__new__(cls)  # pylint: disable=no-value-for-parameter
+        # pylint: disable=attribute-defined-outside-init
```

Comment:
out of curiosity, is there a reason not to fix this?
or is it likely to be in `abc.ABCMeta`?

Reply:
This is due to the typing in this magical method being very weird; logically the line abvoe we know that `obj` is of type `Component`, but asserting that in any way causes different issues. This one I have suppressed for two reasons: firstly, it is the one which the error is most 'true' for (init is not called for another 10 lines or so), secondly it has only just appeared as a detectable thing.

Frankly, this level of messing with magic methods kind of falls outside of the linter's domain

Reply:
@ajCameron found a possibly neater fix for the linting but which masqurades the issue a little more: https://github.com/mewler/mewbot/pull/83/files#diff-2581cf6becc21d8526e55750bf2c59d4c79f271279474ae15b9dbb51158aaca8R21

Reply:
No longer a part of this PR